frontend/src/auth/useAuth.js

```diff
@@ -21,13 +21,59 @@ export const AuthProvider = ({ children }) => {
     window.location.href = '/'
   };
 
+  const postRequestWithImage = (
```

Comment:
Can we just wrap this in a new file/class, as I don't think the place for the image handling should be in auth related codes.